streams/src/test/java/org/apache/kafka/streams/integration/ResetPartitionTimeIntegrationTest.java

```diff
@@ -0,0 +1,193 @@
+import java.util.Optional;
+import java.util.Properties;
+
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.common.serialization.Deserializer;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.common.serialization.Serializer;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.apache.kafka.streams.KafkaStreams;
+import org.apache.kafka.streams.KeyValueTimestamp;
+import org.apache.kafka.streams.StreamsBuilder;
+import org.apache.kafka.streams.StreamsConfig;
+import org.apache.kafka.streams.integration.utils.EmbeddedKafkaCluster;
+import org.apache.kafka.streams.integration.utils.IntegrationTestUtils;
+import org.apache.kafka.streams.kstream.Consumed;
+import org.apache.kafka.streams.processor.TimestampExtractor;
+import org.apache.kafka.test.IntegrationTest;
+import org.apache.kafka.test.TestUtils;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+
+@RunWith(Parameterized.class)
+@Category({IntegrationTest.class})
+public class ResetPartitionTimeIntegrationTest {
+    private static final int NUM_BROKERS = 1;
+    private static final Properties BROKER_CONFIG;
+    static {
+        BROKER_CONFIG = new Properties();
+        BROKER_CONFIG.put("transaction.state.log.replication.factor", (short) 1);
+        BROKER_CONFIG.put("transaction.state.log.min.isr", 1);
+    }
+    @ClassRule
+    public static final EmbeddedKafkaCluster CLUSTER =
+        new EmbeddedKafkaCluster(NUM_BROKERS, BROKER_CONFIG, 0L);
+
+    private static final StringDeserializer STRING_DESERIALIZER = new StringDeserializer();
+    private static final StringSerializer STRING_SERIALIZER = new StringSerializer();
+    private static final Serde<String> STRING_SERDE = Serdes.String();
+    private static final int DEFAULT_TIMEOUT = 100;
+    private final boolean eosEnabled;
+    private static long lastRecordedTimestamp = -2L;
+
+    @Parameters(name = "{index}: eosEnabled={0}")
+    public static Collection<Object[]> parameters() {
+        return asList(
+            new Object[] {false},
+            new Object[] {true}
+        );
+    }
+
+    public ResetPartitionTimeIntegrationTest(final boolean eosEnabled) {
+        this.eosEnabled = eosEnabled;
+    }
+
+    @Test
+    public void testPartitionTimeAfterKStreamReset() throws Exception {
+        final String testId = "-shouldRecoverPartitionTimeAfterReset";
+        final String appId = "appId" + testId;
+        final String input = "input" + testId;
+        final String outputRaw = "output-raw" + testId;
```

Comment:
Why do we need to suffix the appId and topic names with the `testId` ?

Reply:
Just thought it would be a good idea to include more information in the the id names. They can be hardcoded.